### PR TITLE
Add `Array#select`, `Array#reject` and some description.

### PR DESCRIPTION
**Pull request by tebakane**
Add `Array#select`, `Array#reject` and descriptions,
Add spec for `Array#select!` and `Array#reject!`.
Add descriptions for `Array#select!` and `Array#reject!`.

### CODE REVIEW

**Comment by BlaXpirit** [comment]
Doesn't it already have `select`? [Enumerable#select](http://crystal-lang.org/api/Enumerable.html#select%28%26block%3AT-%3E%29-instance-method)

**Comment by tebakane** [comment]
Oh! sorry. i mistook it

**Comment by tebakane** [comment]
I like to add some tests for `Array#select!` and `Array#reject!` for empty array.
And also description for the methods.

File: src/array.cr [hunk 1]
```diff
@@ -862,11 +862,30 @@ class Array(T)
...
+  # ary                                 #=>[2,4]
+  # ary.select!{|elem| elem % 2 == 1 }  #=>[]
+  # ary                                 #=>[]
+  # ```
   def select!
     delete_if { |elem| !(yield elem) }
     self
   end
 
+  # Select array elements that doesn't meet conditions from `self`. Returns `self`.
```
Comment (jhass) [inline src/array.cr]: How about "Removes all elements from `self` for which the given block returns a falsy value"?

File: src/array.cr [hunk 2]
```diff
@@ -862,11 +862,30 @@ class Array(T)
     self
   end
 
+  # Select array elements that meets conditions from `self`. Returns `self`.
```
Comment (jhass) [inline src/array.cr]: Likewise, how about `Removes all elements from `self` for which the given block returns falsy value."?

**Comment by jhass** [comment]
Could you please squash the commits together?

```sh
git remote add upstream https://github.com/manastech/crystal.git
git fetch upstream
git checkout add_select_reject
git rebase -i upstream/master
# Keep the first commit at pick, select squash for any other commit
git push -f origin add_select_reject # assuming you cloned from your fork.
```

Also note `git commit --amend` which allows you to edit the last commit.

**Comment by waj** [comment]
What's the point of adding new tests if the implementation didn't change?

**Comment by tebakane** [comment]
@waj i think we need to know what happen when all the elements of array have been removed. return empty array, nil or exception.

**Comment by waj** [comment]
I understand, is not a big deal. It's just that is a bad practice in TDD to add tests that doesn't fail before changing the implementation.

**Comment by tebakane** [comment]
I'm sorry. I mistook it.
I will commit once more

**Comment by tebakane** [comment]
@waj I understand what you say. Thanks. now I remove the tests. Thanks.
I just add the descriptions. 
I'm sorry for the mishap.

File: src/array.cr [hunk 3]
```diff
@@ -862,11 +862,30 @@ class Array(T)
     self
   end
 
+  # Removes all elements from `self` for which the given block returns falsy value.
+  # 
+  # if no elements found, self become an empty array.
```
Comment (jhass) [inline src/array.cr]: Ah I thought to drop these, it's implicit for me me from the above and also shown in the examples. But if we want to make it explicit I think a better wording would be "If all elements are removed the array becomes empty."

**Comment by jhass** [comment]
Thanks!
